src/interrupt.rs

```diff
@@ -0,0 +1,72 @@
+use alloc::vec::Vec;
+
+#[derive(Debug)]
+pub struct IoApic {
+    pub id: u8,
+    pub address: u32,
+    pub global_system_interrupt_base: u32,
+}
+
+#[derive(Debug)]
+pub enum LocalInterruptLine {
+    Lint0,
```

Comment:
What is `Lint`?

Reply:
Local interrupt I assume. Could rather be LInt though.

Reply:
Yeah, local interrupt. I chose to use `Lint0` and `Lint1` because that matches the Intel Architecture Manual for the APIC

Reply:
I think it makes sense personally given the context of LocalInterrupt.